src/main/java/randoop/generation/DemandDrivenInputCreation.java

```diff
@@ -0,0 +1,489 @@
+   * are compatible with the target type.
+   *
+   * @param typeToIndex a map of types to indices
+   * @param t the target type
+   * @return a list of indices that are compatible with the target type
+   */
+  private static List<Integer> findCompatibleIndices(Map<Type, List<Integer>> typeToIndex, Type t) {
+    List<Integer> compatibleIndices = new ArrayList<>();
+    for (Map.Entry<Type, List<Integer>> entry : typeToIndex.entrySet()) {
+      if (EquivalenceChecker.areEquivalentTypesConsideringBoxing(
+          entry.getKey().getRuntimeClass(), t.getRuntimeClass())) {
+        compatibleIndices.addAll(entry.getValue());
+      }
+    }
+    return compatibleIndices;
+  }
+
+  /**
+   * Executes a set of sequences and updates the sequence collection with each successful execution.
+   * It iterates through each sequence, executes it, and if the execution is normal and yields a
+   * non-null value, the value along with its generating sequence is added or updated in the object
+   * pool.
+   *
+   * @param sequenceCollection the SequenceCollection to be updated with successful execution
+   *     outcomes
+   * @param sequenceSet a set of sequences to be executed
+   */
+  private static void executeAndAddToPool(
+      SequenceCollection sequenceCollection, Set<Sequence> sequenceSet) {
+    for (Sequence genSeq : sequenceSet) {
+      ExecutableSequence eseq = new ExecutableSequence(genSeq);
+      eseq.execute(new DummyVisitor(), new DummyCheckGenerator());
+
+      Object generatedObjectValue = null;
+      ExecutionOutcome outcome = eseq.getResult(eseq.sequence.size() - 1);
+      if (outcome instanceof NormalExecution) {
+        generatedObjectValue = ((NormalExecution) outcome).getRuntimeValue();
+      }
+
+      if (generatedObjectValue != null) {
+        sequenceCollection.add(genSeq);
+      }
+    }
+  }
+
+  /**
+   * Extracts sequences from the sequence collection that can generate an object of the specified
+   * type.
+   *
+   * @param sequenceCollection the SequenceCollection from which to extract sequences
+   * @param t the type of object that the sequences should be able to generate
+   * @return a ListOfLists containing sequences that can generate an object of the specified type
+   */
+  public static SimpleList<Sequence> getCandidateMethodSequences(
+      SequenceCollection sequenceCollection, Type t) {
+    return sequenceCollection.getSequencesForType(t, EXACT_TYPE_MATCH, ONLY_RECEIVERS);
+  }
+
+  /**
+   * Get a subset of the sequence collection that contains sequences that returns specific type of
+   * objects. This method consider boxing equivalence when comparing boxed and unboxed types.
+   *
+   * @param t the type of objects to be included in the subset
+   * @return a list of sequences that contains only the objects of the specified type and their
+   *     sequences
+   */
+  private static SimpleList<Sequence> getSequencesForTypeConsideringBoxing(
```

Comment:
Should this be folded into `SequenceCollection.getSequencesForType()`?

Reply:
TODO.